### PR TITLE
Add error statements to inspect and config library

### PR DESCRIPTION
**Pull request by fearful-symmetry**
## What does this PR do?

The lack of error wrappers in agent lead to a bit of a delay in figuring out https://github.com/elastic/go-ucfg/issues/196.
A few of the statements here weren't even checked for errors, which didn't help.

## Why is it important?

We need improved error handling.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)
- [ ] I have added an integration test or an E2E test

### CODE REVIEW

**Comment by fearful-symmetry** [comment]
Annoyingly, just adding some `fmt.Errorf` statements triggered the test checker...

File: internal/pkg/agent/cmd/inspect.go [hunk 1]
```diff
@@ -130,20 +130,23 @@ type inspectConfigOpts struct {
...
 	if err != nil {
-		return err
+		return fmt.Errorf("error creating logger: %w", err)
 	}
 
 	if !opts.variables && !opts.includeMonitoring {
 		fullCfg, err := operations.LoadFullAgentConfig(ctx, l, cfgPath, true)
 		if err != nil {
-			return err
+			return fmt.Errorf("error loading agent config: %w", err) //I bet it's here
```
Comment (faec) [inline internal/pkg/agent/cmd/inspect.go:139]: Is this a leftover comment?

**Review by faec — APPROVED** [review]
Small nit, looks good

File: internal/pkg/agent/cmd/inspect.go (hunk 1 above)
Comment (fearful-symmetry) [reply internal/pkg/agent/cmd/inspect.go:139]: It's me going mad trying to find https://github.com/elastic/go-ucfg/pull/197, yes

**Comment by cmacknz** [comment]
Force merging, there is no value adding unit tests to detect that our error text wording changed.

**Pull request merged**